### PR TITLE
UIEH-1054: Jest+RTL > Test InternalLink, ProxyDisplay, shouldFocus, TokenDisplay, eHoldings

### PR DESCRIPTION
**Pull request by mariiaschv**
## Purpose

- Add tests for InternalLink, shouldFocus, TokenDisplay components
- Refactoring

Tests for `ProxyDisplay` were written in another [PR](https://github.com/folio-org/ui-eholdings/pull/1386)
For testing `EHoldings` component there was created a separate ticket - [UIEH-1153](https://issues.folio.org/browse/UIEH-1153)

Because of changes in `eholdings.js` and absence of the test file the total coverage is less than expected. The coverage on components other than EHoldings is showing 100%.

image

## Issue
[UIEH-1054](https://issues.folio.org/browse/UIEH-1054)

### CODE REVIEW

**Review by BogdanDenis — APPROVED** [review]

File: src/components/proxy-display.js [hunk 1]
```diff
@@ -6,12 +6,17 @@ import {
   Icon,
 } from '@folio/stripes/components';
 
-export default function ProxyDisplay({ proxy, proxyTypes, inheritedProxyId }) {
+const ProxyDisplay = ({
```
Comment (Vladyslav-Velytskyi) [inline src/components/proxy-display.js:9]: `<ProxyDisplay>` was removed from this directory in the [PR](https://github.com/folio-org/ui-eholdings/pull/1386). Also tests was implemented for it there too. Please remove changes for `<ProxyDisplay>` from the current PR. You could add tests for it if you want, but it's looks like is all covered yet.

File: src/components/should-focus.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { findDOMNode } from 'react-dom';
  * `shouldFocus={true}`, will focus the component's DOM node on mount
  * and on update when the prop is toggled from `false` to `true`.
  */
-export default function shouldFocus(Focusable) {
+const shouldFocus = (Focusable) => {
```
Comment (Vladyslav-Velytskyi) [inline src/components/should-focus.js:10]: Single argument should be without brackets

File: src/components/should-focus.test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    const { getByRole } = renderShouldFocus({ shouldFocus: false });
+
+    const renderedLink = getByRole('link', { name: 'Test link' });
+
+    expect(renderedLink).toBeDefined();
+    expect(renderedLink).not.toHaveFocus();
+  });
+
+  it('should set focus on component after updating props ', () => {
+    const { getByRole, rerender } = render(<MockShouldFocus shouldFocus={false} />);
```
Comment (Vladyslav-Velytskyi) [inline src/components/should-focus.test.js:38]:
```suggestion
    const {
      getByRole,
      rerender,
    } = render(<MockShouldFocus shouldFocus={false} />);
```

File: src/components/token-display.js [hunk 4]
```diff
@@ -1,7 +1,10 @@
 import PropTypes from 'prop-types';
 import { FormattedMessage } from 'react-intl';
 
-export default function TokenDisplay({ token, type }) {
+const TokenDisplay = ({
```
Comment (Vladyslav-Velytskyi) [inline src/components/token-display.js:4]: Please, remove this file with it's test file to `token-display` directory

File: src/components/token-display.js [hunk 5]
```diff
@@ -18,9 +21,14 @@ export default function TokenDisplay({ token, type }) {
...
-}
+};
 
 TokenDisplay.propTypes = {
-  token: PropTypes.object.isRequired,
+  token: PropTypes.shape({
+    prompt: PropTypes.string.isRequired,
+    value: PropTypes.string,
+  }).isRequired,
   type: PropTypes.string.isRequired
```
Comment (Vladyslav-Velytskyi) [inline src/components/token-display.js:31]:
```suggestion
  type: PropTypes.string.isRequired,
```

File: src/eholdings.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+class EHoldings extends Component {
+  static propTypes = {
+    history: ReactRouterPropTypes.history.isRequired,
+    location: ReactRouterPropTypes.location.isRequired,
+    match: ReactRouterPropTypes.match.isRequired,
+    root: PropTypes.shape({
+      addEpic: PropTypes.func.isRequired,
+      addReducer: PropTypes.func.isRequired,
+    }),
```
Comment (Vladyslav-Velytskyi) [inline src/eholdings.js:51]:
```suggestion
    }).isRequired,
```

File: src/eholdings.js [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+    showSettings: PropTypes.bool,
```
Comment (Vladyslav-Velytskyi) [inline src/eholdings.js:52]: Please, add default prop for `showSettings`

File: src/eholdings.js [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+import SettingsRootProxyRoute from './routes/settings-root-proxy';
+import SettingsAccessStatusTypesRoute from './routes/settings-access-status-types';
+import SettingsAssignedUsersRoute from './routes/settings-assigned-users-route';
+import SettingsUsageConsolidationRoute from './routes/settings-usage-consolidation-route';
+
+import KeyShortcutsWrapper from './components/key-shortcuts-wrapper';
+import EHoldingsAppContext from './components/eholdings-app-context';
+import { commands } from './commands';
+
+class EHoldings extends Component {
```
Comment (Vladyslav-Velytskyi) [inline src/eholdings.js:43]: I've suggested to move this file with it's test to separate directory

File: src/eholdings.js [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+      },
+    } = this.props;
+
+    const currentURL = `${pathname}${search}`;
+
+    history.replace(currentURL, {});
+  }
+
+  focusSearchField = () => {
+    const { history, location: { search } } = this.props;
```
Comment (Vladyslav-Velytskyi) [inline src/eholdings.js:77]:
```suggestion
    const {
      history,
      location: { search },
    } = this.props;
```

File: src/eholdings.js [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+                  <Route path={`${rootPath}/notes/new`} exact component={NoteCreate} />
+                  <Route path={`${rootPath}/notes/:noteId`} exact component={NoteView} />
+                  <Route path={`${rootPath}/notes/:id/edit`} exact component={NoteEdit} />
+                  <Route path={`${rootPath}/`} exact component={SearchRoute} />
+                </Switch>
+              </Route>
+            </KeyShortcutsWrapper>
+          </CommandList>
+        </>
+      );
```
Comment (Vladyslav-Velytskyi) [inline src/eholdings.js:132]: We've got two equal wrappers `<CommandList>` for two options. It's better to make some refactoring here.
```suggestion
    return (
      <CommandList commands={commands}>
        {showSettings
          ? (
            <Route path={rootPath} component={SettingsRoute}>
              <Route path={`${rootPath}/knowledge-base/:kbId`} exact component={SettingsKnowledgeBaseRoute} />
              <Route path={`${rootPath}/:kbId/root-proxy`} exact component={SettingsRootProxyRoute} />
              <Route path={`${rootPath}/:kbId/custom-labels`} exact component={SettingsCustomLabelsRoute} />
              <Route path={`${rootPath}/:kbId/access-status-types`} exact component={SettingsAccessStatusTypesRoute} />
              <Route path={`${rootPath}/:kbId/users`} exact component={SettingsAssignedUsersRoute} />
              <Route path={`${rootPath}/:kbId/usage-consolidation`} exact component={SettingsUsageConsolidationRoute} />
            </Route>
          )
         : (
          <>
            <EHoldingsAppContext />
            <KeyShortcutsWrapper focusSearchField={this.focusSearchField}>
              <Route path={rootPath} component={ApplicationRoute}>
                <Switch>
                  <Route path={`${rootPath}/providers/:providerId`} exact component={ProviderShow} />
                  <Route path={`${rootPath}/providers/:providerId/edit`} exact component={ProviderEdit} />
                  <Route path={`${rootPath}/packages/new`} exact component={PackageCreate} />
                  <Route path={`${rootPath}/packages/:packageId`} exact component={PackageShow} />
                  <Route path={`${rootPath}/packages/:packageId/edit`} exact component={PackageEdit} />
                  <Route path={`${rootPath}/titles/new`} exact component={TitleCreate} />
                  <Route path={`${rootPath}/titles/:titleId`} exact component={TitleShow} />
                  <Route path={`${rootPath}/titles/:titleId/edit`} exact component={TitleEdit} />
                  <Route path={`${rootPath}/resources/:id`} exact component={ResourceShow} />
                  <Route path={`${rootPath}/resources/:id/edit`} exact component={ResourceEdit} />
                  <Route path={`${rootPath}/notes/new`} exact component={NoteCreate} />
                  <Route path={`${rootPath}/notes/:noteId`} exact component={NoteView} />
                  <Route path={`${rootPath}/notes/:id/edit`} exact component={NoteEdit} />
                  <Route path={`${rootPath}/`} exact component={SearchRoute} />
                </Switch>
              </Route>
            </KeyShortcutsWrapper>
          </>
        )}
      </CommandList>
    );
```

File: src/eholdings.test.js [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+  );
+
+  afterEach(cleanup);
+  let history = null;
+
+  beforeEach(() => {
+    history = createMemoryHistory();
+  });
+
+  it('should render navigation panel', () => {
```
Comment (Vladyslav-Velytskyi) [inline src/eholdings.test.js:61]: Why do we need this tests if they are commented?

**Review by Vladyslav-Velytskyi — CHANGES_REQUESTED** [review]

File: src/components/token-display/token-display.js [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+import { FormattedMessage } from 'react-intl';
+
+const TokenDisplay = ({
+  token,
+  type,
+}) => {
+  if (token.value) {
+    return (
+      <div data-test-eholdings-details-token={type}>
+        <span>
```
Comment (Vladyslav-Velytskyi) [inline src/components/token-display/token-display.js:11]: Could you also check if this `<span>` is necessary and could it be deleted?

File: src/eholdings.test.js (hunk 11 above)
Comment (mariiaschv) [reply src/eholdings.test.js]: There was problems while testing this component so temporarily tests were commented. Now a separate ticket created to test EHoldings, so this test file is now removed.

File: src/components/token-display/token-display.js [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+          {`${token.prompt}`}
+          :&nbsp;
+          {`${token.value}`}
+        </span>
+      </div>);
+  } else {
+    return (
+      <div data-test-eholdings-details-token-message={type}>
+        <span>
```
Comment (Vladyslav-Velytskyi) [inline src/components/token-display/token-display.js:20]: and this `<span>`

**Review by Vladyslav-Velytskyi — APPROVED** [review]

**Pull request merged**